src/components/AddBoxForm/AddBoxForm.jsx

```diff
@@ -59,21 +71,33 @@ const AddBoxForm = () => {
   });
 
   const [files, setFiles] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  const countryOptions = useMemo(() => countryList().getData(), []);
 
   const onSubmit = async data => {
     const formData = data;
     formData.date = formatDate(data.date);
     formData.launchedOrganically = formData.launchedOrganically === 'yes';
     formData.picture = files.length > 0 ? await uploadBoxPhoto(files[0]) : '';
+    formData.country = formData.country.value;
+    const [latitude, longitude] = await getLatLong(formData.zipcode, formData.country);
 
-    // send form data to server
-    await FYABackend.post('/anchorBox/box', formData, {
-      headers: {
-        'Content-Type': 'application/json',
-      },
-    });
-
-    navigate('/admin');
+    try {
+      setLoading(true);
+      /* eslint-disable object-shorthand */
+      await FYABackend.post('/anchorBox/box', {
+        ...formData,
+        latitude: latitude,
+        longitude: longitude,
+      });
```

Comment:
Can we also set showOnMap to true here? Since all admin-entered boxes should appear on the map.

Reply:
gotcha, then i should also add showOnMap field to the routes in Anchor_Box right?

Reply:
Resolving this since we discussed it in DMs, but yes we should add showOnMap to the AnchorBox routes.